stdlib/test/builtin/test_float_literal.mojo

```diff
@@ -89,6 +108,34 @@ def test_power():
     # assert_almost_equal(FloatLiteral(-4.5) ** -2.5, -0.023279235)
 
 
+def test_mod():
+    assert_equal(FloatLiteral(4.5).__mod__(2), 0.5)
+    assert_equal(FloatLiteral(-4.5).__mod__(2), 1.5)
+    assert_equal(FloatLiteral(6).__mod__(2.5), 1.0)
+
+
+def test_div_mod():
+    pass
+    # TODO: I think this is failing because of a bug in the compiler or int the tuple type:
+    # The test runner is returning the following error when executin `a, b = FloatLiteral(4.5).__divmod__(2.0)`:
+    # invalid call to '__refitem__': callee expects 4 parameters, but 2 were specified
+    # var a: FloatLiteral
+    # var b: FloatLiteral
```

Comment:
Maybe there is a compiler bug, but this can't work for a different reason: `FloatLiteral` is nonmaterializable, so you cannot have a runtime variable with that type. This would be one possible way to work around this:
```mojo
alias PairType = Tuple[FloatLiteral, FloatLiteral]
alias t: PairType = FloatLiteral(4.5).__divmod__(2.0)
assert_equal(t[0], 2.0)  # not sure if direct indexing works on `Tuple` yet, but you get the point
assert_equal(t[1], 0.5)
```

Reply:
It does not work:
```
    alias Pair = Tuple[FloatLiteral, FloatLiteral]
    alias t: Pair = (4.5).__divmod__(2.0)
    assert_equal(t[0], 2.0)
    assert_equal(t[1], 0.5)
```
Gives me the following error:
```
RUN: at line 13: mojo /home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo
+ mojo /home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:124:18: error: invalid call to '__init__': argument #1 cannot be converted from 'Pair' to 'Reference['Tuple[FloatLiteral, FloatLiteral]', ...]
    assert_equal(t[0], 2.0)
                 ^
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:124:18: note: cannot bind an RValue to a reference
    assert_equal(t[0], 2.0)
                 ^
/home/msaelices/src/mojo/stdlib/src/memory/reference.mojo:222:8: note: function declared here
    fn __init__(inout self, value: Self._mlir_type):
       ^
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:125:18: error: invalid call to '__init__': argument #1 cannot be converted from 'Pair' to 'Reference['Tuple[FloatLiteral, FloatLiteral]', ...]
    assert_equal(t[1], 0.5)
                 ^
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:125:18: note: cannot bind an RValue to a reference
    assert_equal(t[1], 0.5)
                 ^
/home/msaelices/src/mojo/stdlib/src/memory/reference.mojo:222:8: note: function declared here
    fn __init__(inout self, value: Self._mlir_type):
```

And, trying to avoid the usage of indexing on Tuples I did this:
```
    alias Pair = Tuple[FloatLiteral, FloatLiteral]
    alias t: Pair = (4.5).__divmod__(2.0)
    assert_equal(t, (2.0, 0.5))
```
Getting this error:
```
RUN: at line 13: mojo /home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo
+ mojo /home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:124:17: error: no matching function in call to 'assert_equal'
    assert_equal(t, (2.0, 0.5))
    ~~~~~~~~~~~~^~~~~~~~~~~~~~~
/home/msaelices/src/mojo/stdlib/src/testing/testing.mojo:118:4: note: candidate not viable: callee expects 1 parameter, but 0 were specified
fn assert_equal[T: Testable](lhs: T, rhs: T, msg: String = "") raises:
   ^
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:124:18: note: failed to infer parameter 'T', argument type 'Tuple[FloatLiteral, FloatLiteral]' does not conform to trait 'Testable'
    assert_equal(t, (2.0, 0.5))
                 ^
/home/msaelices/src/mojo/stdlib/src/testing/testing.mojo:139:4: note: candidate not viable: argument #0 cannot be converted from 'Pair' to 'String'
fn assert_equal(lhs: String, rhs: String, msg: String = "") raises:
   ^
/home/msaelices/src/mojo/stdlib/src/testing/testing.mojo:156:4: note: candidate not viable: callee expects 2 parameters, but 0 were specified
fn assert_equal[
   ^
/home/msaelices/src/mojo/stdlib/test/builtin/test_float_literal.mojo:124:17: note: failed to infer parameter 'type', parameter isn't used in any argument
    assert_equal(t, (2.0, 0.5))
    ~~~~~~~~~~~~^~~~~~~~~~~~~~~
mojo: error: failed to parse the provided Mojo source module
```

Reply:
Can you please try the one I suggested? I.e. instead of comparing the pair, just compare the elements (it's less elegant, but it's what we have right now).

Reply:
I'm gonna take a look at why this is happening.

Reply:
I've figured it out. There was a minor typo in your suggestions. When I saw it again I've found that the second alias was actually a `var` declaration, so it's fixed here: https://github.com/msaelices/mojo/commit/5914b4cf62f50acd7ec5281a2d155b86df4fc7d6

Reply:
Thanks for the `alias Pair` trick, it worked like a charm

Reply:
great, thank you!

Reply:
I'm actually suprised this worked for you: `var t: Pair = (4.5).__divmod__(2.0)` shouldn't compile since `FloatLiteral` is non-materializable. On a related note, I'm running into issues with this in our internal CI. I'll continue to investigate.

Reply:
Yeah, something is wrong with `Tuple` in parameter expressions. I will land your patch with slightly modified tests, and we can circle back later if needed.